### DISCUSSION TITLE
Action on newly created branch

### DISCUSSION THREAD

**Discussion by outrunthewolf** [community/community#26987] [Actions]
<p>We delete our QA branch, and recreate it from develop every week.</p>
<p>We currently use:</p>
<pre><code>on:
   push:
     branches:
       - qa
</code></pre>
<p>But this does not run when we delete the branch and then push a new one back up.</p>
<pre><code>git push -d origin qa
git push origin qa
</code></pre>
<p>Is there anyway I can get QA to build when we re-create it?</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/outrunthewolf">@outrunthewolf</a> ,</p>
<p>I tested with the following steps in my repository ( <strong>myOrg/myRepo</strong> ):<br>
<strong>1)</strong> Set up a workflow like as below in the <strong>master</strong> branch.</p>
<pre><code>name: CI-Dev
on:
  push:
    branches:
      - Dev
jobs:
. . .
</code></pre>
<p><strong>2)</strong> Executed the command “<strong>git clone <a href="https://github.com/myOrg/myRepo.git" rel="nofollow noopener">https://github.com/myOrg/myRepo.git</a></strong>” on my local machine to clone the repository from the remote to the local.</p>
<p><strong>3)</strong> Executed below commands to generate the <strong>Dev</strong> branch on the local based on the <strong>master</strong> branch, then pushed it from local to remote.</p>
<pre><code>git checkout -b Dev
git push origin Dev
</code></pre>
<p><strong>4)</strong> Checked on the web page of the remote repository, I can see the <strong>Dev</strong> branch was pushed to the remote successfully, and the workflow <strong>CI-Dev</strong> was triggered to run.</p>
<p><strong>5)</strong> Then, like as you posted, executed below commands on the local to delete the <strong>Dev</strong> branch from the remote, and re-push the <strong>Dev</strong> branch from local to remote.</p>
<pre><code>git push -d origin Dev
git push origin Dev
</code></pre>
<p><strong>6)</strong> Checked on the web page of the remote repository, the Dev branch was re-pushed to the remote successfully, and the workflow <strong>CI-Dev</strong> also was triggered again to run.</p>
<p><strong>7)</strong> If I added <strong>paths filter</strong> for the push event in the workflow, step <strong>3)</strong> and step <strong>5)</strong> no longer trigger the workflow:</p>
<pre><code>name: CI-Dev
on:
  push:
    branches:
      - Dev
    paths:
      - '**'
jobs:
. . .
</code></pre>
<p>Have you set <strong>paths filter</strong> for the push event in your workflow?</p>
<p>If the problem still exists, please share your repository with us, so that we can check more detailed configurations in your workflow to analyze the root cause.</p>

**Comment by outrunthewolf** [comment]
<p>yes thats right, we do have a paths filter on there!</p>
<pre><code>on:
  push:
    branches:
    - qa
    paths:
    - 'frontend/**'
    - '.github/workflows/build-and-push-frontend-image-qa.yml'
</code></pre>
<p>We have frontend and backend folders in the repo which we watch for changes.</p>
<p>Now you’ve mentioned it, I can understand why that might cause an issue. What’s your advice for that? </p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/outrunthewolf">@outrunthewolf</a> ,</p>
<p>You can try to add the <a href="https://help.github.com/en/actions/reference/events-that-trigger-workflows#create-event-create" rel="nofollow noopener"><strong>Create event</strong></a> in your workflow:</p>
<pre><code>on:
push:
branches:
- qa
  create:
</code></pre>
<p>The  <strong>Create event</strong> will trigger the workflow to run on the created branch.  <strong>P</strong>** ath filters <strong>are not evaluated for</strong> Create event**.</p>
<p>About why the  <strong>p</strong>** ath filters** will prevent the workflow from being triggered when create a new branch, I have reported this question to the appropriate engineering team for further investigation and evaluation. If they have any progress, I will notify you in time, and sometimes the appropriate engineers may directly reply you here.</p>

**Reply by aliok** [reply]
@BrightRan I know it has been 4+ years, but any idea why this is still happening? It would be very nice if we can use `path` and `branches` filters with `push` events here.

**Comment by vaadin-miki** [comment]
<p>You can use <code>on: create</code> and <code>jobs.if</code>:</p>
<pre><code>on: create

jobs:  
job-name:  
if: ${{ (github.event.ref_type == 'branch') &amp;&amp; github.event.ref == 'branch-name' }}
</code></pre>
<p>Side note: it works for me.</p>

**Comment by outrunthewolf** [comment]
<p>Thanks I removed the path filters for now and it operates as expected :+1:</p>
